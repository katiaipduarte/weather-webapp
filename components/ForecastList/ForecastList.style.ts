import styled from 'styled-components'

export const List = styled.section`
	display: flex;
	flex-flow: row;
	color: white;

	.forecast-column {
		margin-right: 2.5rem;
		text-align: center;

		.weekday {
			font-weight: 600;
		}

		.icon-image {
			margin-top: 8px !important;
		}

		.weather-temp {
			font-size: 0.5rem;

			.label {
				font-weight: 500;
				text-transform: uppercase;
				font-size: 0.5rem;
			}
		}
	}

	// width: 50%;
	// margin: 0 auto;
	// border-radius: 5rem;
	// background-color: rgba(0, 0, 0, 0.45);
	// line-height: 1;
	// -webkit-backdrop-filter: blur(10px);
	// backdrop-filter: blur(10px);
	// max-width: 50%;
	// padding: 10rem;
	// margin-top: 30rem;
	// .flex-forecast {
	// 	display: flex;
	// 	justify-content: space-between;
	// 	padding: 5rem 0;
	// 	.weather-forecast-info {
	// 		span {
	// 			margin-left: 10rem;
	// 		}
	// 	}
	// }
`
