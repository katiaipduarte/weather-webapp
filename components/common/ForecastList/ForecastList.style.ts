import styled from 'styled-components'

export const List = styled.section`
	display: grid;
	grid-gap: 1rem;
	grid-template-columns: 1fr 1fr;
	color: white;
	background-color: rgba(0, 0, 0, 0.35);
	width: fit-content;
	padding: 1rem;
	border-radius: 1rem;
	margin: 5rem auto 0;

	.forecast-column {
		align-items: center;
		display: flex;
		justify-content: center;
		cursor: pointer;
		position: relative;

		.weekday {
			font-weight: 400;
		}

		.weather-temp {
			font-size: 0.65rem;

			.label {
				font-weight: 500;
				text-transform: uppercase;
				font-size: 0.5rem;
			}
		}
	}

	@media all and (min-width: 960px) {
		grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
		margin: 2.5rem 2.5rem 0;
		float: left;

		.forecast-column {
			display: block;
			text-align: center;
			.icon-image {
				margin-top: 8px !important;
			}
		}
	}
`
