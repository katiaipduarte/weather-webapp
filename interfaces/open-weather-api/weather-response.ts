import { CurrentWeather } from './current-weather'
import { DailyWeather } from './daily-weather'

export interface WeatherResponse {
	lat: number
	lon: number
	timezone: string
	timezone_offset: number
	current: CurrentWeather
	daily: DailyWeather[]
}
