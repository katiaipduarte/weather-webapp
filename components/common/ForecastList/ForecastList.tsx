import { DailyWeather } from '@interfaces/open-weather-api/daily-weather'
import { WeatherDescription } from '@interfaces/open-weather-api/weather-description'
import { convertUnixTimestampToWeekday } from '@utils/convert-unix-timestamp'
import Image from 'next/image'
import React from 'react'
import { List } from './ForecastList.style'

type Props = {
	forecast: DailyWeather[]
}

const ForecastList = (props: Props) => {
	const { forecast } = props

	const getIconSrc = (weather: WeatherDescription[]): string => {
		return `http://openweathermap.org/img/w/${weather[0].icon}.png`
	}

	const renderList = (): JSX.Element => {
		return (
			<>
				{forecast.map((entry: DailyWeather, i: number) => {
					const symbol = '\u00b0'

					return (
						<React.Fragment key={i}>
							<div className='forecast-column'>
								<p className='weekday'>
									{convertUnixTimestampToWeekday(entry.dt)}
								</p>
								<Image
									src={getIconSrc(entry.weather)}
									width='40px'
									height='40px'
									className='icon-image'
									alt={entry.weather[0].main}
								/>
								<br />
								<span className='weather-temp'>
									<b className='label'>Min: </b>
									{entry.temp.min}
									{symbol}
								</span>
								<br />
								<span className='weather-temp'>
									<b className='label'>Max: </b>
									{entry.temp.max}
									{symbol}
								</span>
							</div>
						</React.Fragment>
					)
				})}
			</>
		)
	}

	return <List aria-label='forecast data'>{renderList()}</List>
}

export default ForecastList
