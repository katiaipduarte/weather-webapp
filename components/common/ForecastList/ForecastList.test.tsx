import { mockWeatherResponse } from '@mocks/weather-response.mock'
import { render } from '@testing-library/react'
import ForecastList from './ForecastList'

describe('<ForecastList /> component Unit Test', () => {
	it('should render 1 <ForecastList /> component', () => {
		const { container } = render(
			<ForecastList forecast={mockWeatherResponse.daily} />
		)
		expect(container).toMatchSnapshot()
	})
})
