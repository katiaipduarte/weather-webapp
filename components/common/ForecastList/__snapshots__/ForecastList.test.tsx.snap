// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastList /> component Unit Test should render 1 <ForecastList /> component 1`] = `
<div>
  <section
    aria-label="forecast data"
    class="sc-bczRLJ dodrXg"
  >
    <div
      class="forecast-column"
    >
      <p
        class="weekday"
      >
        Sunday
      </p>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Clouds"
          class="icon-image"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Min: 
        </b>
        10.58
        °
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Max: 
        </b>
        18.49
        °
      </span>
    </div>
    <div
      class="forecast-column"
    >
      <p
        class="weekday"
      >
        Monday
      </p>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Clouds"
          class="icon-image"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Min: 
        </b>
        12.2
        °
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Max: 
        </b>
        17.79
        °
      </span>
    </div>
    <div
      class="forecast-column"
    >
      <p
        class="weekday"
      >
        Tuesday
      </p>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Clear"
          class="icon-image"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Min: 
        </b>
        10.98
        °
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Max: 
        </b>
        17.3
        °
      </span>
    </div>
    <div
      class="forecast-column"
    >
      <p
        class="weekday"
      >
        Wednesday
      </p>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Rain"
          class="icon-image"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Min: 
        </b>
        10.05
        °
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Max: 
        </b>
        14.65
        °
      </span>
    </div>
    <div
      class="forecast-column"
    >
      <p
        class="weekday"
      >
        Thursday
      </p>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Rain"
          class="icon-image"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Min: 
        </b>
        8.19
        °
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Max: 
        </b>
        16.15
        °
      </span>
    </div>
    <div
      class="forecast-column"
    >
      <p
        class="weekday"
      >
        Friday
      </p>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Rain"
          class="icon-image"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Min: 
        </b>
        8.22
        °
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Max: 
        </b>
        11.77
        °
      </span>
    </div>
    <div
      class="forecast-column"
    >
      <p
        class="weekday"
      >
        Saturday
      </p>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Rain"
          class="icon-image"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Min: 
        </b>
        8.02
        °
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Max: 
        </b>
        15.87
        °
      </span>
    </div>
    <div
      class="forecast-column"
    >
      <p
        class="weekday"
      >
        Sunday
      </p>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Rain"
          class="icon-image"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Min: 
        </b>
        9.92
        °
      </span>
      <br />
      <span
        class="weather-temp"
      >
        <b
          class="label"
        >
          Max: 
        </b>
        17.85
        °
      </span>
    </div>
  </section>
</div>
`;
